hamamatsu.py

```diff
@@ -134,418 +177,402 @@ def set_by_dict(attr, node_text, values):
                     self.enable = enable
                     
                 elif child.tag == "analogGain":
-                    try: 
-                        gain = float(child.text)
+                    try:
+                        gain = int(child.text)
                         assert 0 < gain < 5, ("analogGain must be between 0 "+
                                               " and 5")
-                        self.analogGain = gain
-                    except:
-                        #TODO replace with logger
-                        print("analogGain was given a non-numeric value!")
+                        self.analog_gain = gain
+                    except ValueError as e:  #
+                        # TODO replace with logger
+                        print(f"{e}\n{child.tag} value {child.text} is non-numeric!")
+                        raise
                         
                 elif child.tag == "exposureTime":
                     try: 
                         # can convert scientifically-formatted numbers - good
-                        self.exposureTime = float(child.text)
-                    except:
-                        #TODO: replace with logger
-                        print("exposureTime was given a non-numeric value!")
-                        
+                        self.exposure_time = float(child.text)
+                    except ValueError as e:  #
+                        # TODO replace with logger
+                        print(f"{e}\n{child.tag} value {child.text} is non-numeric!")
+                        raise
+
                 elif child.tag == "EMGain":
-                     try: 
-                        gain = float(child.text)
-                        assert 0 < gain < 255, ("EMGain must be between 0 "+
```

Comment:
Did you mean to remove "0" in the assert text?

Reply:
Whoops, fixed it